rebasehelper/specfile.py

```diff
@@ -1281,9 +1252,8 @@ def _correct_removed_files(self, sources):
                 for f in f_exists:
                     for index, row in enumerate(sec_content):
                         if f in row:
-                            sec_content[index: index+1] = SpecFile.construct_string_with_comment('#' + row)
-                            break
-                self.rpm_sections[key] = (sec_name, sec_content)
+                            del sec_content[index]
```

Comment:
You can't delete items from a list while iterating through it. You have to either iterate backwards, or use list comprehension for example.